### PR TITLE
Add optional hard connection limit to the server.  No more than N con…

### PR DESCRIPTION
**Pull request by scunningham**
…nections will be active at any time.

## What does this PR do?

Adds optional hard limit to the number connections accepted at one time by server.

## Why is it important?

This will limit resource usage on the server.   Both CPU and memory usage is largely driven by the number of connected agents.

## Checklist

- [x ] My code follows the style guidelines of this project
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [x ] I have made corresponding change to the default configuration files
- [x ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Add the 'hard_conn_limit' to the fleet_server
## Related issues

- 

https://github.com/elastic/fleet-server/issues/121

### CODE REVIEW

File: internal/pkg/config/input.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ type Server struct {
 	MaxHeaderByteSize int               `config:"max_header_byte_size"`
 	RateLimitBurst    int               `config:"rate_limit_burst"`
 	RateLimitInterval time.Duration     `config:"rate_limit_interval"`
+	HardConnLimit     int               `config:"hard_conn_limit"`
```
Comment (blakerouse) [inline internal/pkg/config/input.go:60]: Is the `hard_` prefix needed.

What about `max_connections`?

File: internal/pkg/es/mapping.go [hunk 2]
```diff
@@ -1,7 +1,3 @@
-// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
-// or more contributor license agreements. Licensed under the Elastic License;
-// you may not use this file except in compliance with the Elastic License.
```
Comment (aleksmaus) [inline internal/pkg/es/mapping.go:3]: probably will not like removed license header

**Review by aleksmaus — APPROVED** [review]
👍

File: internal/pkg/config/input.go (hunk 1 above)
Comment (scunningham) [reply internal/pkg/config/input.go]: sure.

File: internal/pkg/es/mapping.go (hunk 2 above)
Comment (scunningham) [reply internal/pkg/es/mapping.go:3]: tnx

**Pull request merged**